### ISSUE TITLE
HTMLStyleElement.sheet DOM definition

### ISSUE THREAD

**Issue by lifeiscontent** [microsoft/TypeScript#28098]
**TypeScript Version:**  3.2.0-dev.201xxxxx

**Search Terms:** sheet

**Code**

using the DOM API

```ts
const style = document.querySelector('style');
const sheet = style.sheet;
```

**Expected behavior:**

sheet is returned as a StyleSheet, however ALL browsers do not return a StyleSheet class, they
return a CSSStyleSheet class, so when you do something like sheet.insertRule it
doesn't exist in the type

**Actual behavior:**

style.sheet returns a StyleSheet class

**Playground Link:** https://www.typescriptlang.org/play/index.html#src=const%20style%20%3D%20document.querySelector('style')%3B%0D%0Aconst%20sheet%20%3D%20style.sheet%3B%0D%0A%0D%0Asheet.insertRule('.hello%20%7Bcolor%3A%20%23f00%7D')%3B

**Related Issues:**

**Comment by weswigham** [comment]
Do you still get a CSSStyleSheet when the element you select isn't an HTMLElement (eg, it comes from an xml document)?

**Comment by lifeiscontent** [comment]
@weswigham not sure how to test that. I'd assume the only element that would return a CSSStyleSheet is a style element.

**Comment by saschanaz** [comment]
AFAIK this is based on an assumption that we would ever get a non-CSS stylesheet. Filed w3c/csswg-drafts#3311.

**Comment by SuryaElavazhagan** [comment]
Is there any ways to get writable CSSStyleSheet object from the DOM?

**Comment by lifeiscontent** [comment]
@saschanaz any updates on this?

**Comment by prettydiff** [comment]
I can see on TypeScript version 3.6.3 there is interface **StyleSheet** on line 15526 of file `lib.dom.d.ts` but it does not contain the necessary unique method `insertRule`.

https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule

**Comment by jakebailey** [comment]
Just looking at old issues; this was fixed in #37464.
